File: solutions/86-Ashik/DesignPatternLab/AgeOfVillager/INation.cs
﻿using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace AgeOfVillager
{
    public interface INation
    {
        //A nation has a terrain, house, tree, water resource and so on
        ITree drawTree(MouseEventArgs e);
        IHouse drawHouse(MouseEventArgs e);
        IWaterSource drawWaterSource(MouseEventArgs e);
        Color setColor();
    }
}

File: solutions/58-sakib-2/src/Nations/INation.java
package Nations;

import House.IHouse;
import Terrain.ITerrain;
import Tree.ITree;
import WaterSource.IWaterSource;

public interface INation {
    ITerrain getTerrain();
    ITree getTree();
    IHouse getHouse();
    IWaterSource getWaterSource();
}
